Fix and add unit tests

lewismarshall opened the pull request:
This PR fixes / simplifies core logic and adds some unit tests.

Also an issue with the cmd entry-point issue is resolved for now (by not returning and running as a service after startup).

Currently this PR includes PR #36.

vaijab commented:
Looks like a broken build?

lewismarshall commented:
Drone seems to be broken or running very slow. I'll check when back in...

On Mon, 26 Jun 2017 at 11:44, Vaidas Jablonskis <notifications@github.com>
wrote:

> Looks like a broken build?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UKHomeOffice/keto-k8/pull/39#issuecomment-311025200>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACYpRfagOFgitKI50KCeU1JC1LxJbBeZks5sH4uRgaJpZM4OFBKw>
> .
>

vaijab commented on pkg/kmm/kmm.go:
```diff
@@ -102,6 +102,9 @@ func GetAssets(cfg Config) (err error) {
 		if err = kubeadm.CreateKubeConfig(cfg.KubeadmCfg); err != nil {
 			return err
 		}
+		if err = CreateAndStartKubelet(cfg.KubeadmCfg.CloudProvider, cfg.KubeadmCfg.KubeVersion, true); err != nil {
```
Where is `err` initialized?

vaijab commented on pkg/kmm/kubelet.go:
```diff
@@ -0,0 +1,88 @@
+
+	// Render kubelet.service
+	data := struct {
+		CloudProviderName	string
```
Indentation.

lewismarshall commented:
@vaijab fixed-up your changes in PR #36 and rebased this of that. Fixed the build - was a timeout, added service flag and updated tests to set appropriately.

vaijab commented on pkg/kmm/kmm.go:
```diff
@@ -18,207 +18,267 @@ import (
+// Both structs here use the same config but are bound to different methods...
+
+// Config is tied to the Primary methods (no interface - not for mocking)
+type Config struct {
```
I don't get this? Why embed `ConfigType` struct here?

lewismarshall replied:
Maybe a better name required but essentially there are high level methods (entry-points) and then low-level methods (mockable bits) - all of which use the same config. This allows a simple type cast to move the data...
